Automattic/wp-calypso
# Components: Move placeholder behavior to post item component

This pull request seeks to update the `<PostItem />` component to smartly render with placeholder effects if passed an invalid global post ID. This can be used in displaying a temporary representation of the post before it has been requested successfully from the REST API.

__Testing instructions:__

Verify that the placeholder effect appears the same in this branch as it does in master on the [custom post types listing screen](http://calypso.localhost:3000/types/post) when posts are being requested.

You can also view a standalone version of the component variants on the [DevDocs Blocks screen](http://calypso.localhost:3000/devdocs/blocks/post-item).

__Follow-up Tasks:__

I don't like the `PLACEHOLDER_POST` constant in the `<PostItem />` component and will instead seek to refactor the `<PostRelativeTimeStatus />` component, splitting its behaviors and rendering something useful when post time is not set (e.g. "Unknown", "Forever ago", etc).

/cc @mtias 

Test live: https://calypso.live/?branch=update/cpt-post-item-placeholder
